src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2011,5 +2011,99 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 2;
             return true;
         }
+
+        public static void Reverse(ref byte buf, nuint length)
+        {
+            int numBytes = (int)length;
+            ref byte first = ref buf;
+            ref byte last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -1);
+            int numBytesWritten = 0;
+            if (Avx2.IsSupported && Vector256<byte>.Count * 2 <= numBytes)
+            {
+                Vector256<byte> ReverseMask = Vector256.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, // first 128-bit lane
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);  // second 128-bit lane
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector256<byte>.Count);
```

Comment:
Semantically, using [Subtract](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.subtract?view=net-6.0) would be better.
```suggestion
                last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, numBytes), Vector256<byte>.Count);
```